brave/brave-core
# Extensions protected urls with patch 4928

This fixes brave/brave-browser#4928 and is a working version of #2779. 

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

The webRequest hack can manually tested using https://github.com/fmarier/webrequest-demo:

1. Install the extension via developer mode in `chrome://extensions`.
2. Visit https://sandbox.uphold.com/something and ensure that you're redirected to `brave.com` with the original URL in the query string.
3. Visit https://sandbox.uphold.com/authorize/blah and ensure that you don't get redirected to `brave.com` until Uphold redirects you to `https://sandbox.uphold.com/authorize/404` first.
4. Visit https://api.uphold.com/oauth2/token and ensure that you're not redirected to `brave.com`.

The contentScript hack can be tested using https://github.com/fmarier/contentscript-demo:

1. Install the extension via developer mode in `chrome://extensions`.
2. Visit https://sandbox.uphold.com/something and ensure that you see a popup window.
3. Visit https://sandbox.uphold.com/authorize/blah and ensure that you don't get a popup window until Uphold redirects you to `https://sandbox.uphold.com/authorize/404` first.
4. Visit https://api.uphold.com/oauth2/token and ensure that you don't see a popup window.
5. Repeat steps 2 and 3 with `uphold.com` instead of `sandbox.uphold.com`.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
